nyaa/backend.py

```diff
@@ -49,6 +49,34 @@ def _replace_utf8_values(dict_or_list):
     return did_change
 
 
+def _recursive_dict_iterator(source):
+    ''' Iterates over a given dict, yielding (key, value) pairs,
+        recursing inside any dicts. '''
+    # TODO Make a proper dict-filetree walker
+    for key, value in source.items():
+        yield (key, value)
+
+        if isinstance(value, dict):
+            for kv in _recursive_dict_iterator(value):
```

Comment:
`yield from` does this nicely, too.

Reply:
Good point, I haven't really looked at all the three-isms.